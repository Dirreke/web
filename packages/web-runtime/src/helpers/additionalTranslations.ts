// just a dummy function to trick gettext tools
function $gettext(msg: string): string {
  return msg
}

export const additionalTranslations = {
  fileNotAvailable: $gettext('The requested file is not yet available, please try again later.'),
  activities: $gettext('Activities'),
  noActivities: $gettext('No activities'),
  virusDetectedActivity: $gettext(
    'Virus "%{description}" detected. Please contact your administrator for more information.'
  )
}
