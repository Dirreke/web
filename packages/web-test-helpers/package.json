{
  "name": "web-test-helpers",
  "private": true,
  "peerDependencies": {
      "@vue/composition-api": "1.7.1",
      "vue-gettext": "^2.1.5",
      "vue-router": "3.6.5",
      "vuex": "^3.6.2"
  }
}
