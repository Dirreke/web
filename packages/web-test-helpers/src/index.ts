export * from './mocks/defaultComponentMocks'
export * from './mocks/defaultStubs'
export * from './mocks/axios'

export * from './defaultPlugins'
export * from './helpers'
