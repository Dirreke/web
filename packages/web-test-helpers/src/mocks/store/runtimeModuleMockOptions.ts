export const runtimeModuleMockOptions = {
  runtime: {
    namespaced: true,
    modules: {
      spaces: {
        namespaced: true,
        getters: {
          spaces: jest.fn(() => [])
        },
        mutations: {
          UPDATE_SPACE_FIELD: jest.fn()
        },
        actions: {
          loadSpaceMembers: jest.fn(),
          reloadProjectSpaces: jest.fn()
        }
      }
    }
  }
}
