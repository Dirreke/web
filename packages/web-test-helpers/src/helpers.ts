import { defaultLocalVue } from 'web-test-helpers/src/localVue/defaultLocalVue'
import Vuex, { StoreOptions } from 'vuex'
import { Component } from 'vue'
import { mount as _mount } from '@vue/test-utils'
import { Data, defineComponent, SetupFunction } from '@vue/composition-api'
import { defaultPlugins } from 'web-test-helpers'
export { RouterLinkStub } from '@vue/test-utils'

export const createStore = <T>(storeOptions: StoreOptions<T>) => {
  return {
    install(app) {
      app.prototype.$store = new Vuex.Store(storeOptions)
    }
  }
}

type ComponentType = any

type CompatPlugin = any

type CompatGlobalMountOptions = {
  mocks?: object | false
  stubs?:
    | {
        [key: string]: Component | string | boolean
      }
    | string[]
  attrs?: Record<string, unknown>
  plugins?: (CompatPlugin | [CompatPlugin, ...any[]])[]
  directives?: { [key: string]: any }
  provide?: Record<any, any>
  components?: Record<string, ComponentType>
}

type CompatMountOptions = {
  props?: any
  data?: any
  attachTo?: HTMLElement | string
  global?: CompatGlobalMountOptions
  shallow?: boolean
  slots?: any
}

export const mount = (component: ComponentType, options: CompatMountOptions) => {
  const localVue = defaultLocalVue()

  options?.global?.plugins?.filter((p) => p).forEach(localVue.use)

  for (const [name, component] of Object.entries(options?.global?.components || {})) {
    localVue.component(name, component)
  }

  return _mount(component, {
    localVue,
    ...(options?.shallow && { shouldProxy: options.shallow }),
    ...(options?.props && { propsData: options.props }),
    ...(options?.data && { data: options.data }),
    ...(options?.attachTo && { attachTo: options.attachTo }),
    ...(options?.slots && { slots: options.slots }),
    ...(options?.global?.provide && { provide: options.global.provide }),
    ...(options?.global?.mocks && { mocks: options.global.mocks }),
    ...(options?.global?.stubs && { stubs: options.global.stubs })
  })
}

export const shallowMount = (component: ComponentType, options: CompatMountOptions) => {
  options = options || {}
  options.shallow = true

  return mount(component, options)
}

export const getComposableWrapper = (
  setup: SetupFunction<Data, Data>,
  { mocks = undefined, store = undefined } = {}
) => {
  return mount(
    defineComponent({
      setup,
      template: '<div></div>'
    }),
    {
      global: {
        plugins: [...defaultPlugins(), store],
        ...(mocks && { mocks })
      }
    }
  )
}
