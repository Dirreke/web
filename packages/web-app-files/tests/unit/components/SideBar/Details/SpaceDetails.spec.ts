import SpaceDetails from '../../../../../src/components/SideBar/Details/SpaceDetails.vue'
import stubs from '../../../../../../../tests/unit/stubs'
import { spaceRoleManager, ShareTypes } from 'web-client/src/helpers/share'
import { createStore, defaultPlugins, shallowMount } from 'web-test-helpers'
import { defaultStoreMockOptions } from 'web-test-helpers/src/mocks/store/defaultStoreMockOptions'

const spaceMock = {
  type: 'space',
  name: ' space',
  id: '1',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  spaceQuota: {
    used: 100,
    total: 1000
  }
}

const spaceShare = {
  id: '1',
  shareType: ShareTypes.space.value,
  collaborator: {
    onPremisesSamAccountName: 'Alice',
    displayName: 'alice'
  },
  role: {
    name: spaceRoleManager.name
  }
}

describe('Details SideBar Panel', () => {
  it('displays the details side panel', () => {
    const { wrapper } = createWrapper(spaceMock)
    expect(wrapper).toMatchSnapshot()
  })
})

function createWrapper(spaceResource) {
  const storeOptions = defaultStoreMockOptions
  storeOptions.getters.user.mockImplementation(() => ({ id: 'marie' }))
  storeOptions.modules.runtime.modules.spaces.getters.spaceMembers.mockImplementation(() => [
    spaceShare
  ])
  storeOptions.modules.Files.getters.highlightedFile.mockImplementation(() => spaceResource)
  storeOptions.modules.Files.getters.currentFileOutgoingCollaborators.mockImplementation(() => [
    spaceShare
  ])
  const store = createStore(storeOptions)
  return {
    wrapper: shallowMount(SpaceDetails, {
      global: {
        plugins: [...defaultPlugins(), store],
        stubs,
        directives: {
          OcTooltip: jest.fn()
        },
        provide: {
          displayedItem: spaceResource
        }
      }
    })
  }
}
