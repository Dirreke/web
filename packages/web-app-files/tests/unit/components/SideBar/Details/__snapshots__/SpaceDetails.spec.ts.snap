// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details SideBar Panel displays the details side panel 1`] = `
<div id="oc-space-details-sidebar">
  <div class="oc-space-details-sidebar-image oc-text-center">
    <oc-icon-stub name="layout-grid" filltype="fill" accessiblelabel="" type="span" size="xxlarge" variation="passive" color="" class="space-default-image oc-px-m oc-py-m"></oc-icon-stub>
  </div>
  <div class="oc-flex oc-flex-middle oc-mb-m oc-text-small" style="gap: 15px;">
    <oc-button-stub type="button" size="medium" arialabel="Open member list in share panel" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
      <oc-icon-stub name="group" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
    </oc-button-stub>
    <!---->
    <p>This space has 1 member.</p>
    <oc-button-stub type="button" size="small" arialabel="Open share panel" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium"><span class="oc-text-small">Show</span></oc-button-stub>
  </div>
  <table aria-label="Overview of the information about the selected space" class="details-table">
    <tr>
      <th scope="col" class="oc-pr-s">Last activity</th>
      <td>Invalid DateTime</td>
    </tr>
    <!---->
    <tr>
      <th scope="col" class="oc-pr-s">Manager</th>
      <td><span>alice</span></td>
    </tr>
    <tr>
      <th scope="col" class="oc-pr-s">Quota</th>
      <td>
        <space-quota-stub spacequota="[object Object]"></space-quota-stub>
      </td>
    </tr>
  </table>
</div>
`;
